frontend/src/views/Crops/Crops.tsx

```diff
@@ -22,7 +22,9 @@ import CropsModal from './CropsModal';
 function Crops() {
   const { state, dispatch } = useAppState();
   const navigate = useNavigate();
-  const [fields, setFields] = useState<NMPFileField[]>(state.nmpFile.years[0].fields || []);
+  const [fields, setFields] = useState<NMPFileField[]>(
```

Comment:
Why do we need to clone it here? Any changes should be done through setFields, which makes a new object instead of editing the object in-place. If we're making changes to the base object (and thus messing with the NMP file) elsewhere in this file then that's an issue.